Scripts/modelsystem.py

```diff
@@ -163,6 +167,50 @@ def _add_internal_demand(self, previous_iter_impedance, is_last_iteration):
                         self.dtm.add_demand(demand[mode])
         log.info("Demand calculation completed")
 
+    def _add_external_demand(self,
+                             long_dist_classes = (param.car_classes
+                                 + param.long_distance_transit_classes)):
+                                 # + param.freight_classes
+        if not self.ass_model.use_free_flow_speeds:
+            # If we want to assign all trips with traffic congestion,
+            # then add long-distance trips as background demand
+            zone_numbers = self.ass_model.zone_numbers
+            car_matrices = {}
+            try:
+                # Try getting long-distance trips from separate file
+                with self.long_dist_matrices.open(
+                        "demand", "vrk", zone_numbers,
+                        self.mapping, long_dist_classes) as mtx:
+                    for ass_class in long_dist_classes:
+                        demand = Demand(self.em.purpose, ass_class, mtx[ass_class])
+                        self.dtm.add_demand(demand)
+                        if ass_class in param.car_classes:
+                            car_matrices[ass_class] = demand.matrix
+            except IOError:
```

Comment:
I'd find it better to have user to do this choice when starting model run. I think this way user might not know what assumptions her/his model entails. Also model run might be erroneous without user ever knowing it.

Other option would be to ask user to point out long distance matrices and interrupt if not found. As minimal change solution I would propose `log.warning` message here.

Reply:
Yes, you are right. I now added logging of the events here, but in the next sprint I will make changes to the run-time parameters (we should also change the name of the `--free-flow-assignment` flag to `--long-distance-demand` or similar) and have Maaru do the corresponding updates to the UI.